src/util-memcmp.h

```diff
@@ -152,33 +481,51 @@ static inline int SCMemcmp(const void *s1, const void *s2, size_t len)
 
     return 0;
 }
+#undef SCMEMCMP_BYTES
+#endif
 
+#if defined(__SSE4_1__)
+#include <smmintrin.h>
+#define SCMEMCMP_BYTES 16
 #define UPPER_LOW   0x40 /* "A" - 1 */
 #define UPPER_HIGH  0x5B /* "Z" + 1 */
 
-static inline int SCMemcmpLowercase(const void *s1, const void *s2, size_t len)
+// clang-format off
+static char scmemcmp_sse41_ul[16] __attribute__((aligned(16))) = {
```

Comment:
In general, as someone not that familiar with sse/avx/etc, I'd like to see more comments around what these are for?